commit 671d92a8e764a3266ecab219d791a27fb257be78
Author: David Gross <dgross@google.com>
Date:   Thu Dec 8 21:15:17 2016 +0000

    [DebugInfo] Relax test case for long double constants.
    
    Summary:
    D27549 (partial fix for PR26619) emits a constant value in the debug
    metadata for a floating-point static const that does not exceed 64
    bits in size.  The regression test accompanying that fix assumes that
    a long double exceeds 64 bits in size and hence does not get a
    constant value in the debug metadata.  However, for some targets --
    such as "--target=hexagon-unknown-elf" -- a long double does not
    exceed 64 bits in size, and hence the test fails.
    
    As a temporary fix, modify the regression test to no longer inspect
    the debug metadata for a long double.
    
    Reviewers: cfe-commits, probinson
    
    Differential Revision: https://reviews.llvm.org/D27589
    
    git-svn-id: https://llvm.org/svn/llvm-project/cfe/trunk@289103 91177308-0d34-0410-b5e6-96231b3b80d8

diff --git a/test/CodeGen/debug-info-static-const-fp.c b/test/CodeGen/debug-info-static-const-fp.c
index 64b6309..6ca2d5d 100644
--- a/test/CodeGen/debug-info-static-const-fp.c
+++ b/test/CodeGen/debug-info-static-const-fp.c
@@ -24,4 +24,8 @@ int main() {
 // CHECK: !DIGlobalVariable(name: "dVal", {{.*}}, isLocal: true, isDefinition: true, expr: ![[DEXPR:[0-9]+]]
 // CHECK: ![[DEXPR]] = !DIExpression(DW_OP_constu, 4658387303597904457, DW_OP_stack_value)
 
-// CHECK: !DIGlobalVariable(name: "ldVal", {{.*}}, isLocal: true, isDefinition: true)
+// Temporarily removing this check -- for some targets (such as
+// "--target=hexagon-unknown-elf"), long double does not exceed 64
+// bits, and so we actually do get the constant value (expr) emitted.
+//
+// DO-NOT-CHECK: !DIGlobalVariable(name: "ldVal", {{.*}}, isLocal: true, isDefinition: true)
